File: src/main/java/core/project/ProjectController.java
package core.project;

import core.project.service.ProjectService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
public class ProjectController {
    @Autowired
    ProjectService projectService;

    @RequestMapping(value = {"/", "/projects"}, method = RequestMethod.GET)
    @ResponseBody
    public List<Project> getProjects() {
        return projectService.getProjects();
    }

    @RequestMapping(value = {"/project"}, method = RequestMethod.POST)
    public void createProject(@RequestBody Project project) {
        projectService.createProject(project);
    }

    @RequestMapping(value = {"/project/{id}"}, method = RequestMethod.GET)
    public void readProject(@PathVariable Long id) {
        projectService.getProject(id);
    }

    @RequestMapping(value = {"/project"}, method = RequestMethod.PUT)
    public void updateProject(@RequestBody Project project) {
        projectService.updateProject(project);
    }

    @RequestMapping(value = {"/project/{id}"}, method = RequestMethod.DELETE)
    public void deleteProject(@PathVariable Long id) {
        projectService.deleteProject(id);
    }
}

File: src/main/resources/angular/app/project/project.controller.js
angular.module('project').controller('projectController',
    ['projectService',
        function (projectService) {

    var vm = this;
    vm.getAllProjects = getAllProjects;
    vm.createNewProject = createNewProject;
    vm.deleteProject = deleteProject;

    function getAllProjects() {
        projectService.getAllProjects().then(
                function (d) {
                    vm.projects = d.data;
                }
            );
    }
    
    function createNewProject(project) {
        projectService.createNewProject(project).then(
            function () {
                vm.getAllProjects();
            }
        );
    }

    function deleteProject(project) {
        projectService.deleteProject(project).then(
            function () {
                vm.getAllProjects();
            }
        );
    }

    vm.getAllProjects();
}]);